hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadataBuilder.java

```diff
@@ -235,4 +233,15 @@ private void addToKeysOrFail(
             requiredKeys.add(result.key());
         }
     }
+
+    /**
+     * Checks if the payer key has not been added, and we didn't fail looking it up. No other keys
+     * can be added at this time. This is to maintain the invariant that the payer key is the first
+     * key in requiredKeys list.
+     *
+     * @return true if payer key is not added, false otherwise
+     */
+    private boolean payerKeyNotAdded() {
+        return status == OK && requiredKeys.isEmpty();
```

Comment:
This seems unsafe to me, even with the invariant of a payer key being the first entry in the list. Maybe it's fine though?